bftengine/include/bftengine/ControlStateManager.hpp

```diff
@@ -14,19 +14,53 @@
 #pragma once
 #include <optional>
 #include "IStateTransfer.hpp"
+#include "ReservedPages.hpp"
+#include "Serializable.h"
+#include "SysConsts.hpp"
 
 namespace bftEngine {
 
-class ControlStateManager {
+static constexpr uint32_t ControlHandlerStateManagerReservedPagesIndex = 1;
+static constexpr uint32_t ControlHandlerStateManagerNumOfReservedPages = 1;
+class ControlStateManager : public ResPagesClient<ControlStateManager,
+                                                  ControlHandlerStateManagerReservedPagesIndex,
+                                                  ControlHandlerStateManagerNumOfReservedPages> {
  public:
-  void setStopAtNextCheckpoint();
-  std::optional<uint64_t> getStopCheckpointToStopAt();
+  void setStopAtNextCheckpoint(int64_t currentSeqNum);
+  std::optional<int64_t> getCheckpointToStopAt();
```

Comment:
We use uint64_t below, but int64_t here. Why?

Reply:
SeqNum is a typedef for int64_t. I will change the other one to be int64_t